src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Sprocs/DisableIndex.sql

```diff
@@ -14,27 +14,37 @@
 --         * index table name
 --     @indexName
 --         * index name
+
+GO
 CREATE PROCEDURE [dbo].[DisableIndex]
     @tableName nvarchar(128),
     @indexName nvarchar(128)
+WITH EXECUTE AS 'dbo'
 AS
-SET NOCOUNT ON;
-SET XACT_ABORT ON;
-SET TRANSACTION ISOLATION LEVEL SERIALIZABLE;
-DECLARE @IsExecuted AS INT;
-SET @IsExecuted = 0;
-BEGIN TRANSACTION;
-IF EXISTS (SELECT *
-           FROM   [sys].[indexes]
-           WHERE  name = @indexName
-                  AND object_id = OBJECT_ID(@tableName)
-                  AND is_disabled = 0)
-    BEGIN
-        DECLARE @Sql AS NVARCHAR (MAX);
-        SET @Sql = N'ALTER INDEX ' + QUOTENAME(@indexName) + N' on ' + @tableName + ' Disable';
-        EXECUTE sp_executesql @Sql;
-        SET @IsExecuted = 1;
-    END
-COMMIT TRANSACTION;
-RETURN @IsExecuted;
+DECLARE @errorTxt as varchar(1000)
+       ,@sql as nvarchar (1000)
+       ,@isDisabled as bit = 0
+       ,@isExecuted as int = 0
+
+IF object_id(@tableName) IS NULL
+BEGIN
+    SET @errorTxt = @tableName +' does not exist or you don''t have permissions.'
+    RAISERROR(@errorTxt, 18, 127)
+END
+
+SELECT TOP 1 @isDisabled = is_disabled FROM sys.indexes WHERE object_id = object_id(@tableName) AND name = @indexName
```

Comment:
Change to this SET @IsDisabled = (SELEC ...)

Reply:
Thanks, updated.